app/questionnaire/questionnaire_schema.py

```diff
@@ -914,6 +822,20 @@ def is_repeating_answer(
             or self.is_answer_dynamic(answer_id)
         )
 
+    def get_list_name_for_answer_id(self, answer_id: str) -> str | None:
+        """
+        Returns the name of the list if the answer is dynamic/repeating and none otherwise
```

Comment:
Worth being more explicit here, `if answer is dynamic, in a repeating block or section.`